src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -78,11 +78,21 @@ private void OnApplicationStopping()
 
         private void OnProcessExit(object sender, EventArgs e)
         {
-            ApplicationLifetime.StopApplication();
-            if(!_shutdownBlock.WaitOne(HostOptions.ShutdownTimeout))
+            try
+            {
+                ApplicationLifetime.StopApplication();
+                if(!_shutdownBlock.WaitOne(HostOptions.ShutdownTimeout))
+                {
+                    Logger.LogInformation("Waiting for the host to be disposed. Ensure all 'IHost' instances are wrapped in 'using' blocks.");
+                }
+            }
+            // When hosting with IIS in-process, we detach the Console handle on main thread exit.
+            // Console.WriteLine may throw here as we are logging to console on ProcessExit.
+            // We catch and ignore all exceptions here. Do not log to Console in thie exception handler.
```

Comment:
Is the Console Logger susceptible to this issue? It should be doing the actual console write from a background thread.

I'm actually not 100% convinced we should completely avoid logging in this situation since IMO loggers generally shouldn't throw (i.e. even when disposed they should just become black holes).

Reply:
It looks like it is susceptible: https://github.com/aspnet/Extensions/blob/master/src/Logging/Logging.Console/src/ConsoleLoggerProcessor.cs#L44

Reply:
Do you prefer me to fix the console logger or do this fix here?

Reply:
Hrrrmmmmmhhhghgghhrhrrmrmrmrrmmmmm....

Console logger I think. Loggers shouldn't throw.

Reply:
It does feel weird to add a try/catch around every log message in the console logger. Could it affect perf?

Reply:
Just put it around the call to `WriteMessage`. Other call sites already `try..catch` it.

As to perf, everything is already pretty littered with `try..catch`es. If they affect perf, that damage is done ;)

Reply:
Actually nvrm, I now realize my original fix wouldn't have helped. Console logs are written on a background thread, so the thread that would have crashed would be the one to crash the process.